lighthouse-core/computed/main-thread-tasks.js

```diff
@@ -81,8 +81,13 @@ class MainThreadTasks {
     const tasks = [];
     /** @type {TaskNode|undefined} */
     let currentTask;
+    let lastTraceTs = 0;
 
     for (const event of mainThreadEvents) {
+      // Keep track of the last trace event timestamp
+      if (event.ts > lastTraceTs) lastTraceTs = event.ts;
+      if (currentTask && currentTask.endTime > lastTraceTs) lastTraceTs = currentTask.endTime;
```

Comment:
rather than doing this (which is a hard to follow with the `currentTask` check and the (I assume) `NaN` comparison), could we either pass in `TraceOfTab.timestamps.traceEnd` or (probably less ideal) replicate the logic from there since the events are sorted by ts?